lib/vrl/core/src/lib.rs

```diff
@@ -2,19 +2,20 @@ pub mod prelude;
 mod runtime;
 
 pub use compiler::{
-    enrichment, function, state, type_def::Index, value, Context, Expression, Function, Program,
-    Target, Value,
+    function, state, type_def::Index, value, Context, Expression, Function, Program, Target, Value,
 };
 pub use diagnostic;
 pub use runtime::{Runtime, RuntimeResult, Terminate};
+use std::any::Any;
 
 /// Compile a given source into the final [`Program`].
 pub fn compile(
     source: &str,
-    enrichment_tables: Box<dyn enrichment::TableSetup>,
     fns: &[Box<dyn Function>],
+    external_context: Vec<Box<dyn Any>>,
```

Comment:
Do we have a reason to make this a `Vec`? I think just `Option<Box<dyn Any>>` would suffice for the current use-case, wouldn't it?

If so, I'd prefer that, we can always change it we have a need for more external context to be added.

Reply:
`Option` would work for now.